include/proteus/LambdaRegistry.hpp

```diff
@@ -7,21 +7,28 @@
 
 #include "proteus/CompilerInterfaceTypes.h"
 #include "proteus/Debug.h"
-#include "proteus/Hashing.hpp"
+#include "proteus/Error.h"
 #include "proteus/Logger.hpp"
 
 namespace proteus {
 
 using namespace llvm;
 
+// The LambdaRegistry stores the unique lambda type symbol and JIT member
+// variables. We assume there is always ONLY ONE pending lambda registration
```

Comment:
Why are we making this assumption?

Reply:
Because we clear(ed) pending JIT variables when `registerLambda` executes, so pushed JIT variables must correspond only in the immediate following`registerLambda`. This update here fixes caching to include lambda captures and codifies that assumption. We can improve in the future but we'll need to discuss.